driver.go

```diff
@@ -140,35 +140,62 @@ func (d *Driver) GetCreateFlags() []mcnflag.Flag {
 	}
 }
 
-func (d Driver) client() (*cloudapi.Client, error) {
-	keyData, err := ioutil.ReadFile(d.TritonKeyPath)
-	if err != nil {
-		return nil, err
-	}
-	userAuth, err := auth.NewAuth(d.TritonAccount, string(keyData), "rsa-sha256")
-	if err != nil {
-		return nil, err
+func (d Driver) client() (*compute.ComputeClient, error) {
+	var signer auth.Signer
+	var err error
+
+	if d.TritonKeyPath == "" {
+		signer, err = auth.NewSSHAgentSigner(d.TritonKeyId, d.TritonAccount)
+		if err != nil {
+			return nil, fmt.Errorf("error Creating SSH Agent Signer: %s", err)
+		}
+	} else {
```

Comment:
This doesn't account for us actually passing the KeyMaterial right? Should we allow that too? We do that in terraform-triton and packer-triton

Reply:
This was definitely worth noting on the PR, as we're trying to move all `triton-go` projects to use the same downstream env vars. It would be nice to replace `SDC_KEY_PATH` with `SDC_KEY_MATERIAL` or deprecate it slowly with a future PR.

At the moment I'm suspect of changing the interface of this project since it has a few dependent users (Rancher I believe).

Reply:
I think Rancher's usage of this will automatically pull in the appropriate values via the documented flags, but existing installs would likely be affected, so doing that migration carefully would be a good idea (probably supporting both for some period of time, even if the old way is hidden somehow -- not sure off the top of my head if `docker-machine` supports hidden flags).

Reply:
I didn't change any flags so they _should_ still be usable the old way. Something to test.

Both key signing forms are operationally identical. The only thing that is different is how the key is read/loaded from the local file system or read out of the `ssh-agent` process. End result is the same, SSH key signing for CloudAPI calls.